test/test_genome.py

```diff
@@ -346,89 +347,118 @@ def test_is_gene_in_output_region(rng_seed):
     assert not genome._is_gene_in_output_region(11)
 
 
-def test_mutate_hidden_region(rng_seed):
+def test_mutation_rate(rng_seed, mutation_rate):
+    n_input = 1
+    n_output = 1
+    n_column = 4
+    n_row = 3
+    genome = cgp.Genome(n_input, n_output, n_column, n_row, None, (cgp.Add, cgp.Sub))
     rng = np.random.RandomState(rng_seed)
-    genome = cgp.Genome(1, 1, 3, 1, None, (cgp.Add, cgp.ConstantFloat))
-    dna = [
-        ID_INPUT_NODE,
-        ID_NON_CODING_GENE,
-        ID_NON_CODING_GENE,
-        1,
-        0,
-        0,
-        1,
-        0,
-        0,
-        0,
-        0,
-        2,
-        ID_OUTPUT_NODE,
-        3,
-        ID_NON_CODING_GENE,
-    ]
-    genome.dna = list(dna)
-    active_regions = cgp.CartesianGraph(genome).determine_active_regions()
+    genome.randomize(rng)
 
-    # mutating any gene in inactive region returns True
-    assert genome._mutate_hidden_region(list(dna), 3, active_regions, rng) is True
-    assert genome._mutate_hidden_region(list(dna), 4, active_regions, rng) is True
-    assert genome._mutate_hidden_region(list(dna), 5, active_regions, rng) is True
+    def count_n_immutable_genes(n_input, n_output, n_row):
+        counter = n_input * 3  # 3 is the arity
+        counter += n_output * 2  # 2 of 3 output genes can not be mutated
+        if n_input == 1:
+            counter += (
+                n_row * 2
+            )  # input gene addresses in the first hidden layer can't be mutated in that case
+        return counter
+
+    def count_mutations(prev_dna, new_dna):
+        counter = 0
+        for idx in range(len(prev_dna)):
+            if prev_dna[idx] != new_dna[idx]:
+                counter += 1
+        return counter
+
+    acceptable_error_interval = 0.05  # Todo: get comment for this value from Jakob and Max
+    n = 1000
+    n_immutable_genes = count_n_immutable_genes(n_input, n_output, n_row)
+    n_mutations_expected = n * mutation_rate * (len(genome.dna) - n_immutable_genes)
+    print(len(genome.dna), n_immutable_genes)
+    n_mutations = 0
+    for _ in range(n):
+        temp_dna = genome.dna
+        genome.mutate(mutation_rate, rng)
+        current_n_mutations = count_mutations(temp_dna, genome.dna)
+        n_mutations += current_n_mutations
+
+    assert abs(n_mutations_expected - n_mutations) < (
+        n_mutations_expected * acceptable_error_interval
+    )
```

Comment:
```suggestion
    n_mutations = []
    for _ in range(n):
        dna_old = genome.dna
        genome.mutate(mutation_rate, rng)
        n_mutations.append(count_mutations(dna_old, genome.dna))

    assert np.mean(n_mutations) == pytest.approx(n_mutations_mean_expected)
    assert np.std(n_mutations) == pytest.approx(n_mutations_std_expected)
```
where you'd compute the mean and standard deviation from the binomial distribution.

Reply:
What should I use as values for `rel` in `pytest.approx()`? The assertion passes with `n=10000` and `rel=0.02` which seems reasonable to me. What do you think?